tests/osquery/test_osquery_api.py

```diff
@@ -1395,3 +1398,29 @@ def test_continue_file_carving_invalid_block_data(self):
         }
         response = self.post_as_json("carver_continue", post_data)
         self.assertEqual(response.status_code, 400)
+
+    def test_legacy_public_urls_are_disabled_on_tests(self):
+        routes = ['enroll', 'config', 'carver_start', 'carver_continue', 'distributed_read', 'distributed_write']
+
+        for route in routes:
+            with self.assertRaises(NoReverseMatch):
+                reverse(f"osquery_public_legacy:{route}")
+            self.assertIsNotNone(reverse(f"osquery_public:{route}"))
+
+    def test_mount_legacy_public_endpoints_flag_is_working(self):
+        url_prefix = "/public"
+        routes = ['enroll', 'config', 'carver_start', 'carver_continue', 'distributed_read', 'distributed_write']
+
+        osquery_conf = settings._collection["apps"]._collection["zentral.contrib.osquery"]
+        osquery_conf._collection["mount_legacy_public_endpoints"] = True
+        urlpatterns_w_legacy = tuple(build_urlpatterns_for_zentral_apps())
+        osquery_conf._collection["mount_legacy_public_endpoints"] = False
+        urlpatterns_wo_legacy = tuple(build_urlpatterns_for_zentral_apps())
+
+        for route in routes:
+            self.assertEqual(
+                reverse(f"osquery_public:{route}", urlconf=urlpatterns_w_legacy),
+                url_prefix + reverse(f"osquery_public_legacy:{route}", urlconf=urlpatterns_w_legacy)
+            )
+            with self.assertRaises(NoReverseMatch):
+                reverse("osquery_public:{route}", args=urlpatterns_wo_legacy)
```

Comment:
Weird that the second argument `args` worked 🤔 
Clearly is wrong. Probably ir raising `NoReverseMatch` and the test is passing.
I'll fix.

Reply:
Oh, you merged 😞

Reply:
Oh I missed this one…

Reply:
This is multiple times wrong. it should be osquery_public_legacy too. It is only failing because the args being the list of all URLs, there is no match.

Reply:
Even worse, the f of the f string is missing, this is why it is raising the exception!